### PR TITLE
Research/CAS further colab/features

### PR DESCRIPTION
**Pull request by kiVts**
# About the pull request
Wooo
This adds super dirt cheap minirocket ammo to CAS which does absolutely nothing on its own but create a smoke screen
instead, this can accept any chemical to create a chemical smoke, No it wont allow for custom explosion chemical to be used.
this allows for 
Mass frontline stimming(this is deserved considering how hard is it to get for research)
Mass xeno disruption, 
Mass xeno armor debuff and so on
Mass marine healing????
phosphorus smoke????? 
can custom fuel be used in this case??? not sure.
this allows for other not research things like mass marine feeding(lmao)
chemicals can be mixed and they can be diluted
 Im still missing sprites(currently using inc, will draw them tmr) and probably lots of *bad* code so don't merge it
This generally simplifies the logistics problems research been facing since cave age
this also will encourage to get better OD levels as the numbers can get *very big* when marine is under smoke of stims, or dilute it with water.

# Explain why it's good for the game

More connections  with research to outside world is good 👍 

# Testing Photographs and Procedure

<details>

<summary>Screenshots & Videos</summary>

I did test this

</details>

# Changelog

:cl: KiVts
add: Added a chemical mini rocket ammo for CAS, Which can be used by research to rapidly deploy chemicals on the battlefield, with help of CAS
/:cl:

### CODE REVIEW

**Comment by LynxSolstice** [comment]
+1 silly idea

File: code/modules/cm_marines/dropship_ammo.dm [hunk 1]
```diff
@@ -396,6 +396,53 @@
...
+/obj/structure/ship_ammo/minirocket/smoke/attackby(obj/item/W, mob/user)
+	switch(state)
+		if(ASSEMBLY_EMPTY)
+			if(istype(W, /obj/item/reagent_container/glass/beaker/large) && W.reagents.total_volume == 120)
+				user.drop_held_item(W)
+				W.forceMove(src)
+				chemical = W.reagents
+				to_chat(user, SPAN_NOTICE("You insert the beaker into chemical recepticle"))
+				state = ASSEMBLY_LOCKED
+				desc = initial(desc) + "\nThere is a chemical in the recepticle."
```
Comment (harryob) [inline code/modules/cm_marines/dropship_ammo.dm:418]: you can do this in the get_examine_text proc for certain states

File: code/modules/cm_marines/dropship_ammo.dm [hunk 2]
```diff
@@ -396,6 +396,53 @@
 	spawn(5)
 		fire_spread(impact, create_cause_data(initial(name), source_mob), 3, 25, 20, "#EE6515")
 
+/obj/structure/ship_ammo/minirocket/smoke
+	name = "Smoke mini rocket stack"
+	desc = "A pack of laser guided Smoke mini rockets. To deploy smoke harmlessly, The rocket explodes in mid air seconds before the impact, specially formed charges leads sharpnel upwards away from the target. The smoke contents can be adjusted by research department. Ask them to learn more."
+	icon_state = "minirocket_inc"
+	point_cost = 100
+	fire_mission_delay = 3
+	var/chemical = null
```
Comment (harryob) [inline code/modules/cm_marines/dropship_ammo.dm:405]: null on destroy

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by kiVts** [comment]
image
sprite

**Comment by kiVts** [comment]
fuick I forgot

File: code/modules/cm_marines/dropship_ammo.dm [hunk 3]
```diff
@@ -396,6 +396,64 @@
...
+
+
+/obj/structure/ship_ammo/minirocket/smoke/attackby(obj/item/W, mob/user)
+	switch(state)
+		if(ASSEMBLY_EMPTY)
+			if(istype(W, /obj/item/reagent_container/glass/beaker/large) && W.reagents.total_volume == 120)
+				user.drop_held_item(W)
+				W.forceMove(src)
+				chemical = W.reagents
+				to_chat(user, SPAN_NOTICE("You insert the beaker into chemical recepticle"))
```
Comment (Zonespace27) [inline code/modules/cm_marines/dropship_ammo.dm:416]:
```suggestion
				to_chat(user, SPAN_NOTICE("You insert the beaker into chemical receptacle."))
```

File: code/modules/cm_marines/dropship_ammo.dm [hunk 4]
```diff
@@ -396,6 +396,64 @@
...
+	point_cost = 100
+	fire_mission_delay = 3
+	var/chemical = null
+	var/state = ASSEMBLY_EMPTY
+
+
+/obj/structure/ship_ammo/minirocket/smoke/attackby(obj/item/W, mob/user)
+	switch(state)
+		if(ASSEMBLY_EMPTY)
+			if(istype(W, /obj/item/reagent_container/glass/beaker/large) && W.reagents.total_volume == 120)
```
Comment (Zonespace27) [inline code/modules/cm_marines/dropship_ammo.dm:412]: why does this need to be very specifically a large beaker with == 120u

File: code/modules/cm_marines/dropship_ammo.dm [hunk 5]
```diff
@@ -396,6 +396,64 @@
...
+
+/obj/structure/ship_ammo/minirocket/smoke/detonate_on(turf/impact)
+	if(chemical != null)
+		var/datum/effect_system/smoke_spread/chem/smoke = new/datum/effect_system/smoke_spread/chem()
+		smoke.set_up(chemical,8,0,impact, null)
+		smoke.start()
+		if(!ammo_count && loc)
+			qdel(src)
+	else if (chemical == null)
+		var/datum/effect_system/smoke_spread/S = new/datum/effect_system/smoke_spread()
```
Comment (Zonespace27) [inline code/modules/cm_marines/dropship_ammo.dm:451]:
```suggestion
		var/datum/effect_system/smoke_spread/smoke = new
```
change other vars too

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/cm_marines/dropship_ammo.dm [hunk 6]
```diff
@@ -396,6 +396,64 @@
...
+/obj/structure/ship_ammo/minirocket/smoke/Destroy()
+	chemical = null
+	return ..()
+
+/obj/structure/ship_ammo/minirocket/smoke/detonate_on(turf/impact)
+	if(chemical != null)
+		var/datum/effect_system/smoke_spread/chem/smoke = new/datum/effect_system/smoke_spread/chem()
+		smoke.set_up(chemical,8,0,impact, null)
+		smoke.start()
+		if(!ammo_count && loc)
```
[the remark this answers is not in the document]
Comment (kiVts) [reply code/modules/cm_marines/dropship_ammo.dm:446]: literally copypasted from all other minirockets????? no? it doesnt work like that? do I remove it?

[the remark this answers is not in the document]
Comment (Zonespace27) [reply code/modules/cm_marines/dropship_ammo.dm:446]: if it's copypasted then sure ig

**Comment by Herotitan772** [comment]
i feel like cas has needed some love for a long long time, this is a good idea, i believe that it should be both an OT and research thing, possibly, research can make the (as you have described) chemical based effects and OT can make more explosive based ammunition.
I dont see this as an OP thing because as is OT is hamstrung by the hard explosive cap (WHICH IS DUMB), which limits the strength of OT explosives (the max strength is pitiful, can BARELY kill a tier 1 with a DIRECT hit from a maxcap mortar. everything else is unable to do enough damage, and the fire maxcap is basically to the level of greenfire from pyro (just without the slowing effect))

File: code/modules/cm_marines/dropship_ammo.dm [hunk 7]
```diff
@@ -396,6 +396,62 @@
...
 		fire_spread(impact, create_cause_data(initial(name), source_mob), 3, 25, 20, "#EE6515")
 
+/obj/structure/ship_ammo/minirocket/smoke
+	name = "smoke mini rocket stack"
+	desc = "A pack of laser guided Smoke mini rockets. To deploy smoke harmlessly, the rocket explodes in mid-air seconds before impact, specially formed charges lead sharapnel upwards away from the target. The smoke contents can be adjusted by research department. Ask them to learn more."
+	icon_state = "minirocket_smk"
+	point_cost = 100 ///price in the fabricator
+	fire_mission_delay = 3
+	var/chemical = null ///chemical(s) in the chemical minirocket
+	var/state = ASSEMBLY_EMPTY /// current assembly state of the chemsmoke minircoket
```
Comment (harryob) [inline code/modules/cm_marines/dropship_ammo.dm:406]:
```suggestion
	/// chemical(s) in the chemical minirocket
	var/chemical
	/// current assembly state of the chemsmoke minircoket
	var/state = ASSEMBLY_EMPTY 
```

**Review by harryob — CHANGES_REQUESTED** [review]
one last bit and we should be good to go

File: code/modules/cm_marines/dropship_ammo.dm (hunk 7 above)
Comment (kiVts) [reply code/modules/cm_marines/dropship_ammo.dm]: done

**Comment by kiVts** [comment]
so there was already unused minirocket sprite meant for flares, so using that instead of drawing a new one. just renamed it for sake for being understandable, doesnt touch anything about sprites anymore

**Comment by Herotitan772** [comment]
Tested this today and have a few problems:
First being that the range is way to small, current is a 1 radius plus sign shaped smoke cloud, it should be at least a 3 radius smoke cloud

Seccond being as segrain said full beakers don’t seem to work (equality check???)

third being the major problem…. It doesn’t work… we tested it with a beaker of bi and Kelo and it did nothing…. Tested with a full volley of direct hits…. I suspect this is a problem with chem smoke in general and not specifically this TM but I could be wrong

**Comment by cm13-github** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CM13 Live Instance

##### Revision
Origin: 6a2cf07b91be24fbcbc651e8a46a8ae6445dd924
Pull Request: 47fd4e96d9b63865f6bd25a97dc77f82d3827d43
Server: 0f1dac4dcb22794ac0797cee8eb9c5b0b7c5e7f8

##### Comment

</details>

**Comment by kiVts** [comment]
> Tested this today and have a few problems: First being that the range is way to small, current is a 1 radius plus sign shaped smoke cloud, it should be at least a 3 radius smoke cloud
> 
> Seccond being as segrain said full beakers don’t seem to work (equality check???)
> 
> third being the major problem…. It doesn’t work… we tested it with a beaker of bi and Kelo and it did nothing…. Tested with a full volley of direct hits…. I suspect this is a problem with chem smoke in general and not specifically this TM but I could be wrong

ughm... yeah there should be an equality check...

**Comment by kiVts** [comment]
> Tested this today and have a few problems: First being that the range is way to small, current is a 1 radius plus sign shaped smoke cloud, it should be at least a 3 radius smoke cloud
> 
> Seccond being as segrain said full beakers don’t seem to work (equality check???)
> 
> third being the major problem…. It doesn’t work… we tested it with a beaker of bi and Kelo and it did nothing…. Tested with a full volley of direct hits…. I suspect this is a problem with chem smoke in general and not specifically this TM but I could be wrong

on top of that can you tell about "doesnt work" part better ?

**Comment by Herotitan772** [comment]
> > Tested this today and have a few problems: First being that the range is way to small, current is a 1 radius plus sign shaped smoke cloud, it should be at least a 3 radius smoke cloud
> > Seccond being as segrain said full beakers don’t seem to work (equality check???)
> > third being the major problem…. It doesn’t work… we tested it with a beaker of bi and Kelo and it did nothing…. Tested with a full volley of direct hits…. I suspect this is a problem with chem smoke in general and not specifically this TM but I could be wrong
> 
> on top of that can you tell about "doesnt work" part better ?

Sorry I should have elaborated

So I put a beaker of kelo and bica (60 units of each in a 120 unit large beaker) in the smoke missile pod and loaded it (no issues apart from having to take 10 units out before we could load the beaker in (equality check)). I then went groundside and we tested firing a full volley at a signal flare I was standing on, the smoke was an orangy colour (so the chems were defiantly in the smoke (we had tested just the smoke rounds with no chems earlier and it was white)) and directly hit me and I stood in it for the entire duration, but sadly there was no effect, I scanned myself with a scanner and I had no chems in my system (I checked whilst I was in the smoke and every ~10 seconds after for about 2 minutes and no change)

I’ll keep testing it to see if I can get it working, I’ll keep you posted

**Comment by kiVts** [comment]
> > > Tested this today and have a few problems: First being that the range is way to small, current is a 1 radius plus sign shaped smoke cloud, it should be at least a 3 radius smoke cloud
> > > Seccond being as segrain said full beakers don’t seem to work (equality check???)
> > > third being the major problem…. It doesn’t work… we tested it with a beaker of bi and Kelo and it did nothing…. Tested with a full volley of direct hits…. I suspect this is a problem with chem smoke in general and not specifically this TM but I could be wrong
> > 
> > 
> > on top of that can you tell about "doesnt work" part better ?
> 
> Sorry I should have elaborated
> 
> So I put a beaker of kelo and bica (60 units of each in a 120 unit large beaker) in the smoke missile pod and loaded it (no issues apart from having to take 10 units out before we could load the beaker in (equality check)). I then went groundside and we tested firing a full volley at a signal flare I was standing on, the smoke was an orangy colour (so the chems were defiantly in the smoke (we had tested just the smoke rounds with no chems earlier and it was white)) and directly hit me and I stood in it for the entire duration, but sadly there was no effect, I scanned myself with a scanner and I had no chems in my system (I checked whilst I was in the smoke and every ~10 seconds after for about 2 minutes and no change)
> 
> I’ll keep testing it to see if I can get it working, I’ll keep you posted

weird, last time I tested it on local CO had all the kind of chemicals in him that I fired.

**Comment by cm13-github** [comment]
#### Test Merge Removed

**Comment by Herotitan772** [comment]
Alrighty sorry for the false information on my part, I just tested it again with a beaker full of bicard and it works a charm!

I am an idiot and didn’t realise I had an integrated gas mask on that’s on me….

so yeah I still believe that it needs to create a bigger cloud (my suggestion is a 3 radius cloud as opposed to the current 1 radius), but in terms of the functionality it works as intended

I also checked and after the recent update to this TM, full 120 unit beakers work

**Comment by kiVts** [comment]
I consider this a failure

**Comment by kiVts** [comment]
how did I not close this muh

**Pull request closed without merging**